### DISCUSSION TITLE
Grouped records

### DISCUSSION THREAD

**Discussion by Sof31t** [nocodb/nocodb#3604] [Features]
Hi everyone,

I am wondering if grouping records in some view is possible, something similar to [airtable](https://support.airtable.com/docs/guide-to-grouped-records).

If not implemented yet, is this something you included in your roadmap?

Thank you

**Comment by leraptor** [comment]
This is a needed feature to completely replace airtable.

**Comment by o1lab** [comment]
We will think about it - given how much love this post has garnered in quick time. 

Original issue is #188 - please upvote with thumbs up there too :)

Fyi, we do have underlying groupby apis already. One of the challenges here is that users are connecting to tables which has million+ rows. And showing group by like in other products is not helpful. So we will have to adapt for the better.

**Reply by MGW-SumathyV** [reply]
We can include other filters for e.g datetime along with group by to make the  user to work on dataset range rather than all the rows.

**Reply by guy-borderless** [reply]
Came here looking for this issue. The groupings view is extremely useful IMO.  

> Fyi, we do have underlying groupby apis already. One of the challenges here is that users are connecting to tables which has million+ rows

In my specific case, this is mainly used for views used by sales and back office workers. Humans use those views so mostly shouldn't include millions of rows anyway. so in my use-case I don't mind a LIMIT to get the awesome grouping view. 

An idea about scaling this in cases where number of returned rows is big might be to show the groupings without the rows and load the grouping's rows on click.

**Comment by kenzomartins** [comment]
This feature would boost adoption
